server/src/main/java/org/elasticsearch/index/query/WildcardQueryBuilder.java

```diff
@@ -185,20 +183,20 @@ public static WildcardQueryBuilder fromXContent(XContentParser parser) throws IO
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
         MappedFieldType fieldType = context.fieldMapper(fieldName);
-        Term term;
+
+        Query query;
         if (fieldType == null) {
-            term = new Term(fieldName, BytesRefs.toBytesRef(value));
+            Term term = new Term(fieldName, BytesRefs.toBytesRef(value));
+            query = new WildcardQuery(term);
```

Comment:
nit: if the field does not exist we could return a `MatchNoDocsQuery` ?

Reply:
This is one thing that has confused me in the past: if a field type doesn't exist, we typically still create a query of the same form (see `TermsQueryBuilder#doToQuery` amongst others).

In any case, maybe I could make this change in a follow-up PR, as I was just hoping for a straight refactor?

Reply:
> This is one thing that has confused me in the past: if a field type doesn't exist, we typically still create a query of the same form (see TermsQueryBuilder#doToQuery amongst other examples).

Yes we don't have a clear policy for this. The reason I prefer the `MatchNoDocsQuery` is that we can fold the reason in the message and if users check the Lucene query through the `_validate` API they can see that this field is not present in the mapping. If we build the same form there is no easy way for the user to understand why a specific query matches no document. Anyway we can discuss this in a follow up, no need to make that change in this pr.